libcxx/test/std/utilities/optional/optional.iterator/begin.pass.cpp

```diff
@@ -53,6 +54,10 @@ constexpr bool tests() {
   assert(test<char>());
   assert(test<const int>());
   assert(test<const char>());
+  assert(test<int&>());
+  assert(test<char&>());
+  assert(test<const int&>());
+  assert(test<const char&>());
```

Comment:
Ditto, if you think it's better not to delay implementing LWG4308, cases for `int (&)[42]` and its friends should also be added.